test/storage-luatest/storage_1_test.lua

```diff
@@ -183,3 +183,36 @@ test_group.test_local_call = function(g)
         box.func.sum:drop()
     end)
 end
+
+test_group.test_ref_with_lookup = function(g)
+    local bid1 = vtest.storage_first_bucket(g.replica_1_a)
+
+    g.replica_1_a:exec(function(bid1)
+        local timeout = 0.1
+        local rid = 42
+        local res, err = ivshard.storage._call(
+            'storage_ref_with_lookup',
+            rid,
+            timeout,
+            {bid1}
+        )
+        ilt.assert_equals(err, nil)
+        ilt.assert_equals(res, {})
+        ivshard.storage._call('storage_unref', rid)
+    end, {bid1})
+
+    g.replica_1_a:exec(function(bid1)
+        local bid_extra = 3001
+        local timeout = 0.1
+        local rid = 42
+        local res, err = ivshard.storage._call(
+            'storage_ref_with_lookup',
+            rid,
+            timeout,
+            {bid1, bid_extra}
+        )
+        ilt.assert_equals(err, nil)
+        ilt.assert_equals(res, {bid_extra})
+        ivshard.storage._call('storage_unref', rid)
```

Comment:
> More than one bucket missing

This one is still not covered.

Reply:
Take a look at the "multiple buckets" section of the `test_ref_with_lookup()`.

Reply:
The `moved` result of `storage_ref_with_lookup()` is always empty or just one bucket, this is my point. I can't see a case when `moved` has more than a single bucket in it. I think this is important to test because by accident now or some time afterwards somebody might break the storage code so it would exit on the first loop iteration, and your tests would still pass.